nexus/src/lib.rs

```diff
@@ -225,6 +225,18 @@ impl Server {
                 log.new(o!("component" => "dropshot_external")),
             )
             .config(config.deployment.dropshot_external.dropshot.clone())
+            .version_policy(dropshot::VersionPolicy::Dynamic(Box::new(
+                dropshot::ClientSpecifiesVersionInHeader::new(
+                    omicron_common::api::VERSION_HEADER,
+                    nexus_external_api::latest_version(),
+                )
+                // Since we don't have control over all clients to the external
+                // API, we allow the api-version header to not be specified
+                // (picking the latest version in that case). However, all
+                // clients that *are* under our control should specify the
+                // api-version header.
+                .on_missing(nexus_external_api::latest_version()),
```

Comment:
Feel free to point me to resources that further describe this. I might just be missing context. What's the behavior when a client specifies a version that differs from the version the API expects? Is a previous version of Nexus used to service that request or is the request sent some 4XX/5XX response? Just trying to get a hold on what our Go SDK should be doing here.

Reply:
Will do!